shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.h

```diff
@@ -0,0 +1,34 @@
+#import <Cocoa/Cocoa.h>
+
+@class FlutterResizeSynchronizer;
+
+@protocol FlutterResizeSynchronizerDelegate
+
+// Invoked on platform thread; Delegate should flush OpenGL context and
+// flip the surfaces
+- (void)resizeSynchronizerCommit:(FlutterResizeSynchronizer*)synchronizer;
+
+@end
+
+// Encapsulates the logic for blocking platform thread during window resize
```

Comment:
1. I'll document it in class comment

2. There is no timeout. No sure if there needs to be one. It seems that if you rely on timeout here, something must have gone wrong already. In which case it's probably better to fail early be locking up, than "sort-of work".

3. `requestCommit` is called out of resize as well. There is another thing `FlutterResizeSynchronizer` does - it ensures that delegate commit method is called on platform thread, while raster thread waits in `requestCommit`. It could be that the class needs more generic name, since it's not just resizing that it synchronizes.

Reply:
Documented [here](https://github.com/flutter/engine/pull/21525/commits/1eb4508315f4fa3c194fa810b5fe599caf2f0886).